From 66f3e746a916c2e8bc20fab3d3d5f18a0d9e665b Mon Sep 17 00:00:00 2001
From: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
Date: Wed, 29 Sep 2021 18:03:10 +0300
Subject: [PATCH 136/322] [FIO internal-squash] watchdog: fix recursive
 dependency

Fix the recursive dependency [1] in Kconfig.

[1]
drivers/watchdog/Kconfig:24:error: recursive dependency detected!
drivers/watchdog/Kconfig:24:	symbol SPL_HW_WATCHDOG is selected by SPL_WDT
drivers/watchdog/Kconfig:253:	symbol SPL_WDT depends on SPL_HW_WATCHDOG

Fixes: f36bface6c ("[FIO internal] watchdog: introduce SPL_HW_WATCHDOG")
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
---
 common/spl/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/common/spl/Kconfig b/common/spl/Kconfig
index df116507532..beab53ca02d 100644
--- a/common/spl/Kconfig
+++ b/common/spl/Kconfig
@@ -1262,7 +1262,6 @@ endif
 
 config SPL_WATCHDOG_SUPPORT
 	bool "Support watchdog drivers"
-	imply SPL_WDT if !SPL_HW_WATCHDOG
 	help
 	  Enable support for watchdog drivers in SPL. A watchdog is
 	  typically a hardware peripheral which can reset the system when it
-- 
2.34.1

